tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -2889,13 +2890,47 @@ SPIRVEmitter::getFinalACSBufferCounter(const Expr *expr) {
 
   // AssocCounter#2: referencing some non-struct field
   llvm::SmallVector<uint32_t, 4> indices;
-  if (const auto *decl = getReferencedDef(
-          collectArrayStructIndices(expr, &indices, /*rawIndex=*/true)))
-    return declIdMapper.getCounterIdAliasPair(decl, &indices);
+
+  const auto *base =
+      collectArrayStructIndices(expr, &indices, /*rawIndex=*/true);
+  const auto *decl =
+      (base && isa<CXXThisExpr>(base))
+          ? getOrCreateDeclForMethodObject(cast<CXXMethodDecl>(curFunction))
+          : getReferencedDef(base);
+  return declIdMapper.getCounterIdAliasPair(decl, &indices);
```

Comment:
return followed by ``return nullptr``!

Reply:
Good catch! Will remove in a following up PR.